frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-form/upload-pvc-form.tsx

```diff
@@ -108,25 +119,65 @@ export const UploadPVCForm: React.FC<UploadPVCFormProps> = ({
   isLoading,
   setIsFileRejected,
   setDisableFormSubmit,
+  scConfigMap,
+  storageClasses,
   ...props
 }) => {
   const { t } = useTranslation();
   const operatingSystems = getTemplateOperatingSystems(commonTemplates);
   const operatingSystemHaveDV = operatingSystems.find(
     (os) => os?.baseImageName && os?.baseImageNamespace,
   );
-  const [accessModeHelp, setAccessModeHelp] = React.useState('Permissions to the mounted drive.');
-  const [allowedAccessModes, setAllowedAccessModes] = React.useState(initialAccessModes);
-  const [storageClass, setStorageClass] = React.useState('');
+  const [storageClassName, setStorageClass] = React.useState('');
   const [pvcName, setPvcName] = React.useState('');
   const [namespace, setNamespace] = React.useState(props.namespace);
-  const [accessMode, setAccessMode] = React.useState('ReadWriteOnce');
+  const [accessMode, setAccessMode] = React.useState('');
+  const [volumeMode, setVolumeMode] = React.useState('');
   const [requestSizeValue, setRequestSizeValue] = React.useState('');
   const [requestSizeUnit, setRequestSizeUnit] = React.useState('Gi');
-  const [storageProvisioner, setStorageProvisioner] = React.useState('');
   const [isGolden, setIsGolden] = React.useState(!!osParam);
   const [os, setOs] = React.useState<OperatingSystemRecord>();
   const [osImageExists, setOsImageExists] = React.useState(false);
+  const defaultSCName = getDefaultStorageClass(storageClasses)?.metadata.name;
+  const updatedStorageClass = storageClasses?.find((sc) => sc.metadata.name === storageClassName);
+  const provisioner = updatedStorageClass?.provisioner || '';
+  let accessModes: string[] =
+    provisionerAccessModeMapping[provisioner] || getAccessModeForProvisioner(provisioner);
```

Comment:
what happens when `provisioner` is `''` ?

Reply:
if `provisioner` has a false value (e.g. '' or null) the `provisionerAccessModeMapping` map will return `undefined` and the `getAccessModeForProvisioner` method will return the default value `[rwo, rwx, rox]`